### PR TITLE
Datasearch: Increase in the number of pokemon generated in /randpoke

### PR DESCRIPTION
**Pull request by Distrib-ps**
…& /randmove

PR for https://www.smogon.com/forums/threads/expand-number-of-mons-moves-that-can-be-generated-in-randpoke-or-randmove.3684529/

### CODE REVIEW

**Comment by KrisXV** [comment]
You need to also fix the error messages

**Comment by mia-pi-git** [comment]
This should probably just be a top-level const that the error messages use as well, for simplicity.

**Comment by AnnikaCodes** [comment]
(You committed something after Mia's comment, but didn't actually implement it.)

**Comment by Distrib-ps** [comment]
> (You committed something after Mia's comment, but didn't actually implement it.)

The error message a was fix ? I did not understand what is being asked >.<

**Comment by AnnikaCodes** [comment]
She's asking you to use a constant (`const`) at the top of the file to store the number of Pokémon that `/randpoke` should generate. That way, you can change it more easily, without having to hardcode it everywhere.

**Comment by Distrib-ps** [comment]
> She's asking you to use a constant (`const`) at the top of the file to store the number of Pokémon that `/randpoke` should generate. That way, you can change it more easily, without having to hardcode it everywhere.

Okay, I’ll do that in the next few days, thx !

File: server/chat-plugins/datasearch.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ type Direction = 'less' | 'greater' | 'equal';
 
 const MAX_PROCESSES = 1;
 const RESULTS_MAX_LENGTH = 10;
+const RESULTS_MAX_GENERATE = 30;
```
Comment (AnnikaCodes) [inline server/chat-plugins/datasearch.ts:53]:
```suggestion
const MAX_RANDOM_RESULTS = 30;
```

File: server/chat-plugins/datasearch.ts [hunk 2]
```diff
@@ -176,7 +177,7 @@ export const commands: Chat.ChatCommands = {
 			if (Number.isInteger(num)) {
 				if (qty) throw new Chat.ErrorMessage("Only specify the number of Pok\u00e9mon Moves once.");
 				qty = num;
-				if (qty < 1 || 15 < qty) throw new Chat.ErrorMessage("Number of random Pok\u00e9mon Moves must be between 1 and 15.");
+				if (qty < 1 || MAX_RANDOM_RESULTS < qty) throw new Chat.ErrorMessage(`Number of random Pok\u00e9mon Moves must be between 1 and ${MAX_RANDOM_RESULTS}.`);
```
Comment (AnnikaCodes) [inline server/chat-plugins/datasearch.ts:180]:
```suggestion
				if (qty < 1 || MAX_RANDOM_RESULTS < qty) {
					throw new Chat.ErrorMessage(`Number of random Pok\u00e9mon Moves must be between 1 and ${MAX_RANDOM_RESULTS}.`);
				}
```

File: server/chat-plugins/datasearch.ts [hunk 3]
```diff
@@ -221,7 +222,7 @@ export const commands: Chat.ChatCommands = {
 			if (Number.isInteger(num)) {
 				if (qty) throw new Chat.ErrorMessage("Only specify the number of Pok\u00e9mon once.");
 				qty = num;
-				if (qty < 1 || 15 < qty) throw new Chat.ErrorMessage("Number of random Pok\u00e9mon must be between 1 and 15.");
+				if (qty < 1 || MAX_RANDOM_RESULTS < qty) throw new Chat.ErrorMessage(`Number of random Pok\u00e9mon must be between 1 and ${MAX_RANDOM_RESULTS}.`);
```
Comment (AnnikaCodes) [inline server/chat-plugins/datasearch.ts:225]:
```suggestion
				if (qty < 1 || MAX_RANDOM_RESULTS < qty) {
					throw new Chat.ErrorMessage(`Number of random Pok\u00e9mon must be between 1 and ${MAX_RANDOM_RESULTS}.`);
				}
```

**Comment by AnnikaCodes** [comment]
Looks good! The check failures are unrelated.

**Pull request merged**